### ISSUE TITLE
Make the background rebalancer retry logic work when hostname or ip of source target change

### ISSUE THREAD

**Issue by JelteF** [citusdata/citus#6594]
The background rebalancer will automaticall retry a failed command. This is very useful when something goes wrong. However, due to the command itself we know for sure that the command will never be successful if the node changes. This can happen when we call citus_update_node after a failover.

Then this command:
```
SELECT pg_catalog.citus_move_shard_placement(102012,'172.19.0.3',5432,'172.19.0.2',5432,'force_logical')
```
will fail constantly with:
```
 ERROR: Moving shards to a non-existing node is not supported            +
 HINT: Add the target node via SELECT citus_add_node('172.19.0.2', 5432);+
 CONTEXT: Citus Background Task Queue Executor: citus/postgres           +
```

To avoid this we need new variants of `citus_move_shard_placement` and `citus_copy_shard_placement`, which take source and target groupids instead of hostname+port combinations. Because that will allow a retry of the same command to succeed just fine after a failover has happened. These new versions of these commands should then also be used by the rebalancer.

**Comment by marcocitus** [comment]
Should we use node IDs instead of group IDs? Mainly for consistency with split APIs. Node IDs also appear in GPIDs (for obvious reasons), so it's becoming the more visible variant of referencing a node.

**Comment by JelteF** [comment]
I understand the desire to be consistent with the split API, and I think it's a fair point. The reason I suggested going for group ids is because after discussions with @CyberDem0n I realised that a missing, but probably desirable feature would be a "promote" functionality of secondary nodes. When we would implement such a feature, then by using nodeids here, we would get the issue described above again (only now after citus_promote instead of citus_update_node). 

I guess the question is what we think is more important, preparation for a possible feature that we might have in the future, or staying consistent with other APIs. In this case I would go for the possible feature, since I consider it fairly likely that we'll want to add it in the future now that we start integrating better with patroni. And since the shard move and shard split apis are pretty low level, the chance for user confusion is pretty low. The main reason for consistency would be to not confuse ourselves.

**Comment by marcocitus** [comment]
It can understand the sentiment, since I've made a similar judgement on multiple occasions (e.g. replication_model), but I regret it almost every time because I vastly underestimated the timeline within which those future features would happen. Conversely, I never regretted choosing API consistency.

Given how similar split and move operations are, using a similar API makes sense to me.

**Comment by agedemenli** [comment]
Can we close this one?

**Comment by JelteF** [comment]
Fixed by #6612
